### PR TITLE
task Product cards

### PR DESCRIPTION
**Pull request by vladgusakovskiy**
- [DEMO LINK](https://vladgusakovskiy.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://vladgusakovskiy.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,72 @@
...
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="anonymous">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="./styles.css">
+  </head>
+  <body>
+    <section class="card" data-qa="card">
+      <img
+        class="card__img"
+        src="images/imac.jpeg"
+        width="160"
```
Comment (Jamal-Amoian) [inline src/index.html:18]: Better is to write in css code file

Comment (Jamal-Amoian) [reply src/index.html:18]: It is iMAC:)

File: src/styles.css [hunk 2]
```diff
@@ -0,0 +1,133 @@
+* {
+  box-sizing: border-box;
+}
+
+body,
+ul,
+p {
+  margin: 0;
+}
```
Comment (Jamal-Amoian) [inline src/styles.css:9]: It's better, if you write the same code in separate file

File: src/styles.css [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  flex-direction: column;
+  width: 200px;
+  padding-left: 0;
+
+  list-style: none;
+  border-radius: var(--border-radius);
+  border: 1px solid #f3f3f3;
+
+  --card-margin: 0 16px 16px;
+  --border-radius: 5px;
```
Comment (Jamal-Amoian) [inline src/styles.css:34]: In my opinion, you should call variables in the body block

File: src/styles.css [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+.card__price {
+  margin: var(--card-margin);
+}
+
+.card__info p:first-child {
+  font-size: 12px;
+  line-height: 18px;
+  font-weight: 500;
+  margin: 0 0 4px;
+}
```
Comment (Jamal-Amoian) [inline src/styles.css:52]: Nesting it's bad, you have to avoid nesting

File: src/index.html [hunk 5]
```diff
@@ -1,13 +1,72 @@
...
+      >
+
+      <div class="card__info">
+        <p>
+          Моноблок APPLE A1419 iMac 27" Retina 5K (MNED2UA/A)
+        </p>
+
+        <p>
+          Код товара: 195434
+        </p>
```
Comment (Jamal-Amoian) [inline src/index.html:30]: Put this tag in the separate class, for avoid nesting in the future

File: src/styles.css [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+.card__info p:last-child {
+  font-size: 10px;
+  line-height: 14px;
+  color: #616070;
+}
```
Comment (Jamal-Amoian) [inline src/styles.css:58]: The same mistake as above

File: src/styles.css [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+.card__price p:first-child {
+  font-size: 12px;
+  color: #616070;
+}
+
+.card__price p:last-child {
+  font-size: 16px;
+  font-weight: 700;
+}
```
Comment (Jamal-Amoian) [inline src/styles.css:105]: Again nesting

**Review by Jamal-Amoian — APPROVED** [review]
In my opinion, this is not bad code, but you should to avoid a nesting in the next time

File: src/index.html (hunk 1 above)
Comment (vladgusakovskiy) [reply src/index.html:18]: @Jamal-Amoian Okey, thanks for the review)

File: src/styles.css (hunk 7 above)
Comment (vladgusakovskiy) [reply src/styles.css:105]: @Jamal-Amoian Okey, thanks for the review)

File: src/styles.css (hunk 6 above)
Comment (dellachaise) [inline src/styles.css:58]: Add classes to this paragraphs as Jamal advised and change selectors in lines 47 and 54 to only classes. For example `card__title`

File: src/styles.css (hunk 7 above)
Comment (dellachaise) [inline src/styles.css:105]: Add classes to this paragraphs as Jamal advised and change selectors in lines 97 and 102 to only classes. For example `card__price-tag`

File: src/styles.css [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  background-position: center;
+}
+
+.stars--4 > .stars__star:nth-child(-n+4) {
+  background-image: url(../src/images/star-active.svg);
+}
+
+.card__price {
+  margin-bottom: 16px;
+}
```
Comment (dellachaise) [inline src/styles.css:95]: Split your css code to different files - BEM blocks.

According to [checklist](https://github.com/mate-academy/layout_product-cards/blob/master/checklist.md)

`11.[FILE STRUCTURE] - You should create separate file for each BEM block`

detailed info how to do it in [this video](https://mate.academy/learn/css-advanced/bem#/video/917)

File: src/index.html [hunk 9]
```diff
@@ -1,13 +1,72 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="./styles.css">
+  </head>
+  <body>
+    <section class="card" data-qa="card">
+      <img
+        class="card__img"
+        src="images/imac.jpeg"
+        width="160"
+        height="134"
```
Comment (dellachaise) [inline src/index.html:19]: As Jamal advised - move this code to css.

File: src/index.html (hunk 5 above)
Comment (dellachaise) [inline src/index.html:30]: You should set this paragraphs as elements of card block.

```suggestion
      <div class="card__info">
        <p class="card__title">
          Моноблок APPLE A1419 iMac 27" Retina 5K (MNED2UA/A)
        </p>

        <p class="card__code">
          Код товара: 195434
        </p>
```

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,72 @@
...
+      <div class="card__price">
+        <p>
+          Цена:
+        </p>
+
+        <p>
+          69 999 грн
+        </p>
+      </div>
+
```
Comment (dellachaise) [inline src/index.html:60]: You should set this paragraphs as elements of card block too. Add classes to them and use this classes in css to style.

**Review by dellachaise — CHANGES_REQUESTED** [review]
Nise start 👍 

Why don't you change your code according to Jamal review?
You should change it or write in comments your opinion if you think another way.

Please, add fixes according to comments.

Before sending task to review please check it according to  [checklist](https://github.com/mate-academy/layout_product-cards/blob/master/checklist.md)

You always could find checklist in a task description.
Screenshot 2022-07-19 at 10 03 05

**Review by dellachaise — APPROVED** [review]
Great job 🔥

**Pull request closed without merging**